packages/qdrant-loader/src/qdrant_loader/connectors/confluence/connector.py

```
@@ -792,162 +804,256 @@ def _clean_html(self, html: str) -> str:
+    async def fetch_by_id(self, entity_id: str) -> Document | None:
+        """Fetch a single Confluence content item by its content ID."""
+        try:
+            params = {"expand": _CONTENT_EXPAND}
+            safe_entity_id = quote(entity_id, safe="")
+            content = await self._make_request(
+                "GET", f"content/{safe_entity_id}", params=params
+            )
+            if not content or not self._should_process_content(content):
+                return None
+            return self._process_content(content, clean_html=True)
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)